drivers/i2c/i2c_shell.c

```diff
@@ -270,6 +270,36 @@ static int cmd_i2c_read(const struct shell *shell_ctx, size_t argc, char **argv)
 	return ret;
 }
 
+/* i2c speed <device> <speed>
+ * For: speed see constants like I2C_SPEED_STANDARD
+ */
+static int cmd_i2c_speed(const struct shell *shell_ctx, size_t argc, char **argv)
+{
+	char *s_dev_name = argv[ARGV_DEV];
+	const struct device *dev;
+	uint32_t dev_config = 0;
+	uint32_t speed;
+	int ret;
+
+	dev = device_get_binding(s_dev_name);
+	if (!dev) {
+		shell_error(shell_ctx, "I2C: Device driver %s not found.",
+			    s_dev_name);
+		return -ENODEV;
+	}
+
+	speed = strtol(argv[ARGV_DEV + 1], NULL, 10);
+	dev_config = I2C_SPEED_SET(speed);
+
+	ret = i2c_configure(dev, dev_config);
```

Comment:
I think it would be safer to initialize `dev_config` with `i2c_get_config()`. At least `I2C_MODE_CONTROLLER` seems to be missing in the value configured here.